### ISSUE TITLE
RFE: new option monitor-session-bell, to bell on events in other session's windows

### ISSUE THREAD

**Issue by m1027** [tmux/tmux#3882]
Hi,

I'd appreciate very much the following enhancement. I am not aware such thing exists already.

Current situation:

On a bell, the according window is highlighted in the status line (see man tmux, option monitor-bell). However, if I have multiple sessions open, I am not alarmed when there are event's in _the other session's window_.

Enhancement:

So, I imagine  a new option like _monitor-session-bell_ to enable/disable highlighting the session number/name in the very left of the status bar, for any event in another session's window.

Advantage:

I can keep working in the main session and do not have to cycle through all other sessions and windows.

Variant:

There is a related issue to this: Even if working in a single session, you might miss an alarm if there are _too many windows open_ that do not fit the status line. So, this RFE could also be slightly tweaked to trigger a session-bell on _any_ alarm (even from the own session's windows).

Alternative:

Writing an alert-bell hook.

Thanks

**Comment by nicm** [comment]
I'm not entirely certain I've understood everything you are asking for, but if you just wanted to show in the status line if any session has a bell, you could do something like:
~~~~
#{m/r:1,#{S:#{W:#{window_bell_flag}}}}
~~~~
I think you actually want to show if any session *except the attached one*, which is a bit more tricky because `S` does not have an "attached session" argument like `W`... something like this works for me, although its a but ugly:
~~~~
#{?#{m/r:^$,#{s/#{session_name}//:#{S:#{W:#{?#{window_bell_flag},#{session_name},}}}}},0,1}
~~~~
So you could put something in status-left like:
~~~~
tmux set -g status-left '#{?#{m/r:^$,#{s/#{session_name}//:#{S:#{W:#{?#{window_bell_flag},#{session_name},}}}}},,#[bg=red]}[#{session_name}]#[default] '
~~~~
However, I'm not opposed to adding a simpler format variable, something like `other_session_bell_flag` (and similar for activity and silence). If you want to look at that, `format.c` is where to go.

**Comment by nicm** [comment]
Actually, I have ignored my own usual advice here. Using session IDs will be more reliable:
~~~~
tmux set -g status-left '#{?#{m/r:^$,#{s/#{session_id} //:#{S:#{W:#{?#{window_bell_flag},#{session_id} ,}}}}},,#[bg=red]}[#{session_name}]#[default] '

**Comment by nicm** [comment]
Hmm no session IDs don't work because of the $... adding some spaces as delimiters makes session_name better but even it will have problems if the session name contains regexp special characters:
~~~~
tmux set -g status-left '#{?#{m/r:^$,#{s/ #{session_name} //:#{S:#{W:#{?#{window_bell_flag}, #{session_name} ,}}}}},,#[bg=red]}[#{session_name}]#[default] '
~~~~
To do this reliably in all cases I think we either need a dedicated format, or to extend `S` to have a different format for the active session so it can be easily skipped.

**Comment by m1027** [comment]
Thanks for all the cool ideas.

Yes, you got me right. The main goal is achieved, when I have a chance to get notified whenever there is a bell from a window of _any_ session, including the own.

(An bonus upgrade to this was to just get notified from _any other_ session, but not my own session. But let's leave this case aside for now.)

So, for the first, simpler case I tried your inital config example:

- Added to .tmux.conf: `set -g status-left '#{m/r:1,#{S:#{W:#{window_bell_flag}}}} '`
- tmux opened
- Two winodws created
- In window 2, I inserted executed: `sleep 5; printf "\a"`
- Then I switched to window 1 (before printf was called)
- After 5 seconds: Indeed I the little alarm was triggerd (changed from 0 to 1)

Cool.

Then I tried with another session:
- In open tmux I created _another_ session
- I opened again two windows (in this _other_ session), and inserted again sleep 5; printf "\a"` in window 2.
- Then I switched to window 1 (in the other session), then I switched to my inital session (all before printf was called)
- After 5 seconds: The alarm was NOT triggerd (stayed 0)

This was kind of unexpected.

- So, I switched again to the _other_ session, window 1, to see what happened there: The window 2 got the alarm, and, by this switch, the new little alarm changed from 0 to 1 (!)
- Even when I switched back to the original session, the alarm stayed at 1 (triggered)

So, as a summary:

The above code for tmux.conf does unfortunatelly not work when the bell is activated in the _other_ session. Only swiching between the sessions kind of updates the state.

Any idea how we could achieve the goal to get notified on _any_ bell of _any_ session with pure config tweaks?

**Comment by nicm** [comment]
Ah yes it probably won't update the status line when other sessions change. What if you set `status-interval` to 1, does it make it update?

**Comment by m1027** [comment]
Yeah! Alltogether that is actually what I was looking for. Thanks!

Summary for those having the same question:

To get a visual trigger in case of _any_ bell from _any_ session, you may want to add this to your tmux.conf:
- `set -g status-left '[#{m/r:1,#{S:#{W:#{window_bell_flag}}}}] '`
- `set -g status-interval 1`
